src/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/HostWorkspace/LanguageServerProjectLoader.cs

```diff
@@ -181,6 +181,14 @@ protected sealed record RemoteProjectLoadResult(RemoteProjectFile ProjectFile, P
     protected abstract Task<RemoteProjectLoadResult?> TryLoadProjectInMSBuildHostAsync(
         BuildHostProcessManager buildHostProcessManager, string projectPath, CancellationToken cancellationToken);
 
+    /// <summary>Called after a project is unloaded to allow the subtype to clean up any resources associated with the project.</summary>
+    /// <remarks>
+    /// Note that this refers to unloading of the project on the project-system level.
+    /// So, for example, changing the target frameworks of a project, or transitioning between
+    /// "file-based program" and "true miscellaneous file", will not result in this being called.
+    /// </remarks>
+    protected abstract ValueTask OnProjectUnloadedAsync(string projectFilePath);
```

Comment:
Why is this needed? Isn't it the derived type already the one calling Unload, so why can't it just call it's logic directly?